### PR TITLE
Support DPP 0.23

### PR DESCRIPTION
**Pull request by HashEngineering**
## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added or updated relevant unit/integration/functional/e2e tests
- [x] I have made corresponding changes to the documentation

**For repository code-owners and collaborators only**
- [ ] I have assigned this pull request to a milestone

### CODE REVIEW

File: core/src/main/java/org/bitcoinj/wallet/DerivationPathFactory.java [hunk 1]
```diff
@@ -68,7 +69,23 @@ public ImmutableList<ChildNumber> blockchainIdentityECDSADerivationPath() {
                 .add(ChildNumber.ZERO_HARDENED) //sub feature 0
                 .add(ChildNumber.ZERO_HARDENED) //key type (0 is ECDSA key)
                 .add(ChildNumber.ZERO_HARDENED) //identity index (default to 0 for now)
-                .add(ChildNumber.ZERO_HARDENED) //key index (default to 0 for now)
+                //.add(ChildNumber.ZERO_HARDENED) //key index (default to 0 for now)
```
Comment (HashEngineering) [inline core/src/main/java/org/bitcoinj/wallet/DerivationPathFactory.java:72]: this comment needs to be removed and the documentation above this function is out of date.

File: core/src/main/java/org/bitcoinj/wallet/DerivationPathFactory.java [hunk 2]
```diff
@@ -68,7 +69,23 @@ public ImmutableList<ChildNumber> blockchainIdentityECDSADerivationPath() {
...
+        return ImmutableList.<ChildNumber>builder()
+                .add(FEATURE_PURPOSE)
+                .add(coinType)
+                .add(FEATURE_PURPOSE_IDENTITIES)
+                .add(ChildNumber.ZERO_HARDENED) //sub feature 0
+                .add(ChildNumber.ZERO_HARDENED) //key type (0 is ECDSA key)
+                .add(ChildNumber.ZERO_HARDENED) //identity index (default to 0 for now)
+                .add(new ChildNumber(index, true)) //key index (default to 0 for now)
                 .build();
     }
```
Comment (HashEngineering) [inline core/src/main/java/org/bitcoinj/wallet/DerivationPathFactory.java:90]: is this new method even used?

**Comment by HashEngineering** [comment]
Unit tests are failing, need to fix.

File: core/src/main/java/org/bitcoinj/wallet/DerivationPathFactory.java [hunk 3]
```diff
@@ -56,35 +56,66 @@ public ImmutableList<ChildNumber> identityInvitationFundingDerivationPath() {
...
+     * m/9'/5'/5'/0'/1'/(blockchain identity index = 0) (mainnet)
+     * m/9'/1'/5'/0'/1'/(blockchain identity index = 0) (testnet, devnets)
+     */
+    public ImmutableList<ChildNumber> blockchainIdentityBLSDerivationPath() {
+        return ImmutableList.<ChildNumber>builder()
+                .add(FEATURE_PURPOSE)
+                .add(coinType)
+                .add(FEATURE_PURPOSE_IDENTITIES)
+                .add(ChildNumber.ZERO_HARDENED) //sub feature 0
+                .add(ChildNumber.ONE_HARDENED) //key type (0 is ECDSA key)
```
Comment (Syn-McJ) [inline core/src/main/java/org/bitcoinj/wallet/DerivationPathFactory.java:101]: Should this comment here still say `0 is ECDSA key`?

**Review by Syn-McJ — APPROVED** [review]
Looks good

File: core/src/main/java/org/bitcoinj/wallet/DerivationPathFactory.java (hunk 3 above)
Comment (HashEngineering) [reply core/src/main/java/org/bitcoinj/wallet/DerivationPathFactory.java:101]: The comment should say `(1 is BLS key)`

**Pull request merged**